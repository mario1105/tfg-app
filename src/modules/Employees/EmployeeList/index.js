import React, { useEffect, useState } from 'react'
import * as R from 'ramda'
import EmployeeList from "./EmployeeList"
import employeesService from "../../../services/employeesService"

const EmployeeListContainer = ({ role }) => {
    const [employees, setEmployees] = useState([])

    const handleRemoveEmployee = (id) => setEmployees(R.filter((employee) => employee.id !== id, employees))

    useEffect(() => {
        const fetchData = async () => {
            const response = await employeesService.getEmployees()
            setEmployees(response)
        }
        fetchData()
    }, [])

    return (
        <EmployeeList role={role} employees={employees} handleRemoveEmployee={handleRemoveEmployee} />
    )
}

export default EmployeeListContainer

