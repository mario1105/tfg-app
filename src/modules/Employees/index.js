import React from 'react'
import {
    Typography, makeStyles, Button
} from '@material-ui/core';
import ReturnButton from "../../components/ReturnButton";
import EmployeeList from "./EmployeeList";
const useStyles = makeStyles(theme => ({
    employeesBox: {
        backgroundColor: theme.app.background['300'],
        padding: theme.spacing(7),
        borderRadius: 10,
        margin: '10em 20em 10em 20em',
        display: 'flex',
        flexDirection: 'column'
    }
}))

const Employees = ({ location }) => {
    const user = location.state

    const classes = useStyles()

    return (
        <div className={classes.employeesBox}>
            <EmployeeList />
            <ReturnButton previousRoute="dashboard" user={user} style={{ marginTop: '2em' }} />
        </div>
    )
}

export default Employees

