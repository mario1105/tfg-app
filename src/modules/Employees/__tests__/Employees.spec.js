import React from 'react'
import { shallow } from 'enzyme'
import Employees from "../../Employees"

const renderComponent = () => {
    return (shallow(<Employees location={{state: {}}} />))
}

let wrapper
describe('Employees Unit Tests', () => {
    it('renders as expected', () => {
        wrapper = renderComponent()
        expect(wrapper).toMatchSnapshot()
    })
})
