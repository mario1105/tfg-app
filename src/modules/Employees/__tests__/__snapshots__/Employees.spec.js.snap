// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employees Unit Tests renders as expected 1`] = `
<div
  className="makeStyles-employeesBox-1"
>
  <WithStyles(ForwardRef(Typography))
    style={
      Object {
        "color": "white ",
      }
    }
    variant="h5"
  >
    Employee List
  </WithStyles(ForwardRef(Typography))>
  <EmployeeListContainer />
  <ReturnButton
    previousRoute="dashboard"
    style={
      Object {
        "marginTop": "2em",
      }
    }
    user={Object {}}
  />
</div>
`;
